### PR TITLE
404 page dashboard - Route 404 errors to haml page

### PR DESCRIPTION
**Pull request by nkiruka**
# Description
Part of the prerequisite to localizing the strings on the 404 page in Dashboard is to ensure all 404 errors are routed to the same 404 page (not_found.html.haml).

- Summary of change
Add a new route
Override setting in application_controller to use the new custom route
All 404 errors are routed to the new  "not_found.html.haml" view.  
Styles were added to errors.scss
Default 404.html page was removed
New strings are added to en.yml for translation

## Links
- screen shot
Screen Shot 2019-10-04 at 3 52 37 PM

- [jira](https://codedotorg.atlassian.net/secure/RapidBoard.jspa?rapidView=28&projectKey=FND&modal=detail&selectedIssue=FND-769)

## Testing story
Manually tested existing routes that call the render_404 method.

# Reviewer Checklist:

- [ ] Tests provide adequate coverage
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: dashboard/config/locales/en.yml [hunk 1]
```diff
@@ -1166,3 +1166,7 @@ en:
     privacy_policy: 'privacy policy'
     logout: 'Logout'
     yes: 'Yes'
+  page_not_found_title: "Page Not Found"
+  page_not_found_header: '404: Page Not Found'
+  page_not_found_description: "The page you are looking for doesn't exist at this address. Check the web address and resubmit. Or, go back to our homepage."
+  page_not_found_button: "hey"
```
Comment (Hamms) [inline dashboard/config/locales/en.yml]: I assume this is supposed to be
```suggestion
  page_not_found_button: "Return home"
```

File: dashboard/app/views/errors/not_found.html.haml [hunk 2]
```diff
@@ -0,0 +1,11 @@
+.error-parent
+  .error-child
+    %img{src: '/shared/images/sad-bee-avatar.png'}
```
Comment (Hamms) [inline dashboard/app/views/errors/not_found.html.haml]:
```suggestion
    %img{src: '/shared/images/sad-bee-avatar.png', alt: I18n.t(:page_not_found_title)}
```

**Review by Hamms — COMMENTED** [review]
I'm getting the following error when visiting `/404` or when hitting a legit 404:

ActionController::RoutingError at /404
======================================

> uninitialized constant ErrorsController

actionpack (5.0.7.2) lib/action_dispatch/routing/route_set.rb, line 46
----------------------------------------------------------------------

``` ruby
   41         private
   42   
   43           def controller(req)
   44             req.controller_class
   45           rescue NameError => e
>  46             raise ActionController::RoutingError, e.message, e.backtrace
   47           end
   48   
   49           def dispatch(controller, action, req, res)
   50             controller.dispatch(action, req, res)
   51           end
```

File: dashboard/app/views/errors/not_found.html.haml (hunk 2 above)
Comment (Hamms) [reply dashboard/app/views/errors/not_found.html.haml]: this still needs to be resolved

File: dashboard/config/locales/en.yml (hunk 1 above)
Comment (Hamms) [reply dashboard/config/locales/en.yml]: this still needs to be resolved

**Review by Hamms — COMMENTED** [review]
Still have some outstanding comments from my last review; LGTM after those are resolved

File: dashboard/app/views/errors/not_found.html.haml (hunk 2 above)
Comment (nkiruka) [reply dashboard/app/views/errors/not_found.html.haml]: thanks!  I resolved but I had conflicts with circle.yml files that had been removed and forgot to git stash pop the changes I made.

**Review by Hamms — APPROVED** [review]

File: dashboard/app/assets/stylesheets/errors.scss [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+  color: $purple;
+  font-size: 28px;
+  font-family: 'Gotham 7r', sans-serif;
+  margin-top: 50px;
+  margin-bottom: 30px;
+  line-height: 1.2em;
+}
+
+button {
+  -webkit-appearance: none;
```
Comment (breville) [inline dashboard/app/assets/stylesheets/errors.scss:23]: Do we get appropriate button styling already without adding new CSS here?

**Pull request merged**